khronicle-core/src/androidUnitTest/kotlin/StackTraceTagGeneratorTests.kt

```diff
@@ -0,0 +1,42 @@
+package com.juul.khronicle
+
+import java.util.function.Supplier
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class StackTraceTagGeneratorTests {
+
+    @Test
+    fun tagMatchesClassName() {
+        val generatedTag = StackTraceTagGenerator.getTag()
+        assertEquals(StackTraceTagGeneratorTests::class.java.simpleName, generatedTag)
+    }
+
+    @Test
+    fun tagStripsAnonymousClassNumber() {
+        val supplier = Supplier<String> { StackTraceTagGenerator.getTag() }
+
+        /* Examples of `class.java.name`:
+         *
+         * | Kotlin | supplier::class.java.name                                                                     |
+         * |--------|-----------------------------------------------------------------------------------------------|
+         * | 1.4.31 | com.juul.khronicle.StackTraceTagGeneratorTests$tagStripsAnonymousClassNumber$supplier$1 |
+         * | 1.5.10 | com.juul.khronicle.StackTraceTagGeneratorTests$$Lambda$4/1813123723                     |
```

Comment:
nitpick: I think this is copy/pasted, but would be nice to fix the formatting while we're in here.

```suggestion
         * | Kotlin | supplier::class.java.name                                                               |
         * |--------|-----------------------------------------------------------------------------------------|
         * | 1.4.31 | com.juul.khronicle.StackTraceTagGeneratorTests$tagStripsAnonymousClassNumber$supplier$1 |
         * | 1.5.10 | com.juul.khronicle.StackTraceTagGeneratorTests$$Lambda$4/1813123723                     |
```